connectors/src/connectors/slack/temporal/workflows.ts

```diff
@@ -88,6 +89,7 @@ export async function syncOneChannel(
   fromTs: number | null
 ) {
   console.log(`Syncing channel ${channelName} (${channelId})`);
+  await joinChannel(parseInt(connectorId, 10), channelId);
```

Comment:
You're not really checking that this worked here. Returning an error and checking the error would guarantee that you act based on what really happened (as per my comment to change the signature of this function)

Reply:
I took the decision to let the next part of the code fail, because this is what we really care about.
I see this line as a "try to join if you can ,let the rest handle the errors".

Reply:
I think it's quite hard to understand when reading

Reply:
I will add a comment or make it explicit.

Reply:
I really really think we should handle error cases when they arise.

Reply:
I agree 100%, I felt this case was slightly different, it was a "try to join".
I wonder if there is not cases where we can sync a channel where the bot is not invited, but that's probably me just overthinking it.

I will add error handling.